agency/README.md

```diff
@@ -326,19 +347,83 @@ Path Params:
 
 If `stop_id` is specified, `GET` will return an array with a single stop record, otherwise it will be a list of all stop records.
 
+## Reservation Type
+The reservation type enum expresses the urgency of a given reservation. This can be useful when attempting to quantify metrics around trips: for example, computing passenger wait-time. In the `on_demand` case, passenger wait-time may be quantified by the delta between the `reservation_time`, and the pick-up time; however, in the `scheduled` case, the wait time may be quantified based on the delta between the `scheduled_trip_start_time` found in the Trips payload, and the actual `trip_start_time`. 
+| `reservation_type` | Description                                                            |
+|--------------------|------------------------------------------------------------------------|
+| `on_demand`        | The passenger requested the vehicle as soon as possible                |
+| `scheduled`        | The passenger requested the vehicle for a scheduled time in the future |
+
+## Reservation Method
+The reservation method enum describes the different ways in which a passenger can create their reservation.
+| `reservation_method` | Description                                               |
+|----------------------|-----------------------------------------------------------|
+| `app`                | Reservation was made through an application (mobile/web)  |
+| `street_hail`        | Reservation was made by the passenger hailing the vehicle |
+| `phone_dispatch`     | Reservation was made by calling the dispatch operator     |
+
+## Fare
+The Fare object describes a fare for a Trip. 
+| Field           | Type                  | Required/Optional | Field Description                                                                       |
+|-----------------|-----------------------|-------------------|-----------------------------------------------------------------------------------------|
+| quoted_cost     | Float                 | Required          | Cost quoted to the customer at the time of booking                                      |
+| actual_cost     | Float                 | Required          | Actual cost after a trip was completed                                                  |
+| components      | `{ [string]: float }` | Optional          | Breakdown of the different fees that composed a fare, e.g. tolls                        |
+| currency        | string                | Required          | ISO 4217 currency code                                                                  |
+| payment_methods | `string[]`            | Optional          | Breakdown of different payment methods used for a trip, e.g. cash, card, equity_program |
+
+## Trips
```

Comment:
From the new modes guidance draft: `Because the MDS Agency API does not have a top-level endpoint for handling trip data, the trip_attributes field can be optionally provided on the /event endpoint for trip_end events.`

For Taxi we determined that trip_end didn’t actually capture all of the use cases — you could have a trip leave the jurisdiction and still need to get addl. trip info. This is why we decided to add a Trips (but really trip metadata) endpoint for agency.

Reply:
I am a bit concerned that this is so different from the [provider `trips` endpoint](https://github.com/openmobilityfoundation/mobility-data-specification/tree/main/provider#trips).  This seems likely to lead to confusion, and implementation headaches.

Also, for provider trips we don't track any reservation-related information, so adding that in would likely require some nontrivial implementation changes.  Right now our trips data is based on listening for ride starts and ride ends and the telemetry in-between; all of these things have an associated trip ID already so we just group things that share a trip ID and ignore things without a trip ID when creating a trip object.  A reservation event does not have a trip ID though, so associating reservation info to a trip takes more work than that.

Reply:
Great feedback, thank you!

Originally, I'd actually called this 'Trip Metadata', as that's really what it is; it's not a comprehensive object which represents a Trip (while in Provider, there are single objects for Trips), rather, it's additional metadata needed for regulatory purposes which doesn't fit in well with events/telemetry. Some more alignment with Provider trips could work well.

As far as reservations not necessarily having a `trip_id`, though this is something that it is not required by the Agency API for micromobility, we find quite frequently in the reservation events we receive. I think that for Taxi/TNC is pretty important to have a `trip_id` associated with reservations when considering the entire vehicle journey (while this is not necessarily the case for micromobility).

As a whole, these changes really should not impact Micromobility operations substantially. I think the main information for Trip Metadata we'd care about for Micromobility, would be `distance` (odometer distance is much more accurate than trying to string together GPS points).